cli/cli_app.go

```diff
@@ -223,6 +224,47 @@ func runTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 	return runTerragruntWithConfig(terragruntOptions, terragruntConfig, false)
 }
 
+func processHooks(hooks []config.Hook, terragruntConfig *config.TerragruntConfig, terragruntOptions *options.TerragruntOptions, previousExecError ...error) error {
+	if len(hooks) == 0 {
+		return nil
+	}
+
+	errorsOccurred := []error{}
+
+	terragruntOptions.Logger.Printf("Detected %d Hooks", len(hooks))
+
+	for _, curHook := range hooks {
+		allPreviousErrors := append(previousExecError, errorsOccurred...)
+		if shouldRunHook(curHook, terragruntOptions, allPreviousErrors...) {
+			terragruntOptions.Logger.Printf("Executing hook: %s", curHook.Name)
+			actionToExecute := curHook.Execute[0]
+			if actionToExecute != "" {
```

Comment:
Prob better to validate this stuff during parsing, not part way through execution.